### DISCUSSION TITLE
파일 분리 질문드립니다!

### DISCUSSION THREAD

**Discussion by oneulddu** [storycraft/node-kakao#333] [Q&A]
안녕하세요 
node-kakao를 통하여 처음 node.js 에 접하게 되었습니다.
그러다보니 미숙한 부분이 많아 질문 드리고 싶은게 있는데요

pm2를 통하여 파일이 변경될 때 자동으로 리로드가 되는 걸 하고 싶어 
watch 옵션을 통해 넣었으나, 이 부분의 문제는 수정을 할 때마다 로그인을 다시 해야되는
문제가 생겼습니다. 
그래서 main.js 에서 로그인을 처리하고 sub.js 에서 기능을 만들어 기능을 추가하고 
리로르 하려고 시도했으나 
main.js 한 파일 가지고만 하면 잘 되던 기능이 sub.js 를 만들어
 로그인 부분을 제외한 기능 부분만 넣으면 main.js 와 서로 별개의 파일이 되는 거 같습니다.

main.js 에서 로그인을 하고 sub.js 에서 client.on 과 같이 기능을 사용하려면 어떤 별도의 
조치가 필요한지 질문 드립니다 ㅠㅠ

**Comment by raravel** [comment]
파일을 분리한다고 해서 각각의 파일이 다른 프로세스로 돌아가는 게 아닙니다.
`main.js` 와 `sub.js`를 분리한다고 하여도 `sub.js`를 수정하면 전부 재시작합니다.

더 쉽게 표현하자면, 카카오톡 PC버전 프로그램을 파일수정시마다 계속 재시작하는 거라고 보시면 됩니다.
